### DISCUSSION TITLE
SSL validation failed

### DISCUSSION THREAD

**Discussion by maruthi8** [boto/botocore#2840] [Q&A] [answered]
While trying to read objects from s3 it raise exception saying  `[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1131)`

This occurs only if I connect to cloudflare zero trust.
`resource = boto3.resource('s3')
bucket = AwsConfig.AWS_STORAGE_BUCKET_NAME
key="s3 file location"
resource.ObjectSummary(bucket, key).storage_class`

**Comment by RyanFitzSimmonsAK** [comment, marked as answer]
Hi @maruthi8, thanks for reaching out. There is a section on the `SSL: CERTIFICATE_VERIFY_FAILED` error in the AWS CLI [troubleshooting guide](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-troubleshooting.html#tshoot-certificate-verify-failed). The possible causes it mentions there are either that the AWS CLI doesn't trust your proxy's certificate or your configuration isn't pointing to the correct CA root certificate location. Please refer to the troubleshooting steps to resolve this issue and let us know if you have any follow up questions.
